code/java/pipelines-workshop/src/main/java/com/myorg/PipelineStack.java

```diff
@@ -38,39 +41,47 @@ public WorkshopPipelineStack(final Construct parent, final String id, final Stac
         final CdkPipeline pipeline = CdkPipeline.Builder.create(this, "Pipeline")
             .pipelineName("WorkshopPipeline")
             .cloudAssemblyArtifact(cloudAssemblyArtifact)
-            
+
             // Generates the source artifact from the repo we created in the last step
             .sourceAction(CodeCommitSourceAction.Builder.create()
                 .actionName("CodeCommit") // Any Git-based source control
                 .output(sourceArtifact) // Indicates where the artifact is stored
                 .repository(repo) // Designates the repo to draw code from
                 .build())
-            
+
                 // Builds our source code outlined above into a could assembly artifact
             .synthAction(SimpleSynthAction.Builder.create()
-                .installCommands(List.of("npm install -g aws-cdk")) // Commands to run before build
+                .installCommands(Arrays.asList("npm install -g aws-cdk")) // Commands to run before build
                 .synthCommand("npx cdk synth") // Synth command (always same)
                 .sourceArtifact(sourceArtifact) // Where to get source code to build
                 .cloudAssemblyArtifact(cloudAssemblyArtifact) // Where to place built source
-                .buildCommands(List.of("mvn package")) // Language-specific build cmds
+                .buildCommands(Arrays.asList("mvn package")) // Language-specific build cmds
                 .build())
             .build();
 
-        final WorkshopPipelineStage deploy = new WorkshopPipelineStage(this, "Deploy");
+        final PipelineStage deploy = new PipelineStage(this, "Deploy");
         final CdkStage deployStage = pipeline.addApplicationStage(deploy);
+
+        final Map<String, StackOutput> TestViewerEndpointOutputs = new HashMap<>();
+        TestViewerEndpointOutputs.put("ENDPOINT_URL", pipeline.stackOutput(deploy.hcViewerUrl));
+
         deployStage.addActions(ShellScriptAction.Builder.create()
             .actionName("TestViewerEndpoint")
-            .useOutputs(Map.of("ENDPOINT_URL", deploy.hcViewerUrl))
-            .commands(List.of("curl -Ssf $ENDPOINT_URL"))
```

Comment:
`Map.of` and `List.of` are not supported in Java 8 which is the minimum Java version we support.

To fix this I replaced..

-  `Map.of` with `new HashMap`
- `List.of` with `Arrays.asList`

There was also an existing type error that was being thrown on the `useOutputs()` param. To fix this I replaced `deploy.hcEndpoint` with `pipeline.stackOutput(deploy.hcEndpoint)` to get the appropriate type object that is expected in `useOutputs()`.